application/features/bootstrap/UnitTestsContext.php

```diff
@@ -60,22 +63,25 @@ public function purgeDatabase()
      * username, they will be deleted.
      *
      * @param string $username
+     * @param string[] $properties
      * @return User
      */
-    protected function createNewUserInDatabase($username)
+    protected function createNewUserInDatabase($username, $properties = [])
     {
         $existingUser = User::findByUsername($username);
```

Comment:
It might be worth overriding the `$username` with `$properties['username']` (if set), to protect against conflicting username data in the `$username` and `$properties` parameters. Otherwise, this initial check for (and removal of) any conflicting user won't be checking the correct username.

Something like this would probably work:
`$username = $properties['username'] ?? $username`

Reply:
Interesting observation. Would that go against the description in the method comment? "Create a new user in the database with the given username..." Maybe it would be better to unset a username in the properties array instead. What do you think?

Reply:
Hmm, good catch. It would probably be best, then, to either assert that the `$properties` array doesn't contain a 'username' or add a note to the description that a 'username' in the `$properties` will override the value in the `$username` parameter. Either way seems fine.